ddl/column.go

```diff
@@ -861,7 +861,7 @@ func (w *worker) onModifyColumn(d *ddlCtx, t *meta.Meta, job *model.Job) (ver in
 		// Since column type change is implemented as adding a new column then substituting the old one.
 		// Case exists when update-where statement fetch a NULL for not-null column without any default data,
 		// it will errors.
-		// So we set zero original default value here to prevent this error. besides, in insert & update records,
+		// So we set zero original default value here to prevent this error. Besides, in insert & update records,
```

Comment:
For line 867, I think we should use this logic
```
prior 1: if new col has its default value, using it as the origin-default value.
prior 2: if old col has its origin-default value, use it as the origin-default value.
prior 3: use the zero value (partial logic in `generateOriginDefaultValue`) 
```

Reply:
Otherwise, issue 24951's update statement if happens in write-only / write-reorg state, it may have the same error

Reply:
This logic after I told you, I've tried (TestChangingColOriginDefaultValueAfterAddCol) there will be no problem, so there is no change, do you have any repetition of the case?

Reply:
Well, it's probably the type of columns in the test that caused the problem